tests/test_cli.py

```diff
@@ -170,10 +170,12 @@ def test_logout_creds_dne(runner, mocked_creds_path):
     (["temp.fa", "temp2.fa"], True),
 ])
 def test_standard_uploads(runner, upload_mocks, files, threads):
```

Comment:
Would be nice to have a few test cases `@pytest.mark.parametrize` around uploading various metadata keys/values and coercion of those into the right types/key names. This could be as tests of the internal functions I'm suggesting we move into a `helpers.py` file somewhere above.